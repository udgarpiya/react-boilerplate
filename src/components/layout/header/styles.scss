.header-component {
  &__main-container {
    padding: 0 15px;
    height: 50px;
    justify-content: center;
    display: flex;
    align-items: center;
    position: sticky;
    z-index: 1;
    top: -1px;
    left: 0;
    box-shadow: 0px 0px 5px 1px rgba(0, 0, 0, 0.15);
  }
}