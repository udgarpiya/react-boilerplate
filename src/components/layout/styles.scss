.layout-component {
  &__children-container {
    min-height: calc(100vh - 100px);
  }
}