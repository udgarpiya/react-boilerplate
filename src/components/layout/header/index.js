import React from 'react'
import './styles.scss'

const Header = () => {
  return (
    <div className="header-component__main-container">
      <h1>Header Component</h1>
    </div>
  )
}

export { Header }
export default Header
