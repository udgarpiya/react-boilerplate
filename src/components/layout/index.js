import React from 'react'
import PropTypes from 'prop-types'

import Header from './header'
import Footer from './footer'
import CustomHead from '../custom-head'
import './styles.scss'

const Layout = ({ children, title, description }) => {
  return (
    <div className="layout-component__main-component">
      <CustomHead title={title} description={description} />
      <Header />
      <div className="layout-component__children-container">{children}</div>
      <Footer />
    </div>
  )
}

Layout.defaultProps = {
  title: '',
  description: '',
}

Layout.propTypes = {
  children: PropTypes.node.isRequired,
  title: PropTypes.string,
  description: PropTypes.string,
}

export { Layout }
export default Layout
