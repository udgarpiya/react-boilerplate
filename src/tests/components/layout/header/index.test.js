import React from 'react'
import { shallow } from 'enzyme'
import { Header } from 'components/layout/header'

describe('Header Component', () => {
  it('It should have main component and icon', () => {
    const component = shallow(<Header />)

    const HeaderMain = component.find('.header-component__main-container')

    expect(HeaderMain.length).toBe(1)
    expect(component).toMatchSnapshot()
  })
})
