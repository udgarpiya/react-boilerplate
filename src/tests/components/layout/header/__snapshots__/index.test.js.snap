// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component It should have main component and icon 1`] = `
<div
  className="header-component__main-container"
>
  <h1>
    Header Component
  </h1>
</div>
`;
